File: Python/khudi/material/Material.java
package khudi.material;

import khudi.color.RGBColor;

/**
 * @section DESCRIPTION
 *
 * The Material Class.
 * Provides the Material class ....
 *
 */
public class Material
{
	private double gloss;
	private double transparency;
	private double reflection;
	private double refractionIn;
	private double refractionOut;
	private RGBColor color;

	public Material()
	{
		gloss = 2.0;
		transparency = 0.0;
		reflection = 0.0;
		refractionIn = 0.0;
		refractionOut = 0.0;
		color = new RGBColor(0.0, 0.0, 0.0);
	}

	// specifies the Gloss (or shininess) of the element
	// value must be between 1 (very shiney) and 5 (matt) for a realistic effect 
	public double GetGloss()
	{
		return gloss;
	}

	public void SetGloss(double value)
	{
		gloss = value;
	}

	// defines the transparency of the element. 
	// values must be between 0 (opaque) and 1 (fully transparent);
	public double GetTransparency()
	{
		return transparency;
	}

	public void SetTransparency(double value)
	{
		transparency = value;
	}

	// specifies how much light the element will reflect
	// value must be between 0 (no reflection) to 1 (total reflection/mirror)
	public double GetReflection()
	{
		return reflection;
	}

	// refraction index
	// specifies how the material will bend the light rays
	// value must be between <0,1] (total reflection/mirror)
	public void SetReflection(double value)
	{
		reflection = value;
	}

	public double GetRefractionIn()
	{
		return refractionIn;
	}

	public void SetRefractionIn(double value)
	{
		refractionIn = value;
	}

	public double GetRefractionOut()
	{
		return refractionOut;
	}

	public void SetRefractionOut(double value)
	{
		refractionOut = value;
	}

	// indicates that the material has a texture and therefore the exact
	// u,v coordinates are to be calculated by the element
	// and passed on in the GetColor function
	public boolean HasTexture()
	{
		return false;
	}

	// retrieves the actual color of the material
	public RGBColor GetColor()
	{
		return color;
	}

	public void SetColor(RGBColor colorp)
	{
		color = colorp;
	}

}

File: Python/khudi/material/Material.py
#!/usr/bin/env python
""" generated source for module Material """
from __future__ import print_function
# package: khudi.material
# 
#  * @section DESCRIPTION
#  *
#  * The Material Class.
#  * Provides the Material class ....
#  *
#
from khudi.color.RGBColor import RGBColor


class Material(object):
    """ generated source for class Material """
    gloss = float()
    transparency = float()
    reflection = float()
    refractionIn = float()
    refractionOut = float()
    color = None

    def __init__(self):
        """ generated source for method __init__ """
        self.gloss = 2.0
        self.transparency = 0.0
        self.reflection = 0.0
        self.refractionIn = 0.0
        self.refractionOut = 0.0
        self.color = RGBColor(0.0, 0.0, 0.0)

    #  specifies the Gloss (or shininess) of the element
    #  value must be between 1 (very shiney) and 5 (matt) for a realistic effect 
    def GetGloss(self):
        """ generated source for method GetGloss """
        return self.gloss

    def SetGloss(self, value):
        """ generated source for method SetGloss """
        self.gloss = value

    #  defines the transparency of the element. 
    #  values must be between 0 (opaque) and 1 (fully transparent);
    def GetTransparency(self):
        """ generated source for method GetTransparency """
        return self.transparency

    def SetTransparency(self, value):
        """ generated source for method SetTransparency """
        self.transparency = value

    #  specifies how much light the element will reflect
    #  value must be between 0 (no reflection) to 1 (total reflection/mirror)
    def GetReflection(self):
        """ generated source for method GetReflection """
        return self.reflection

    #  refraction index
    #  specifies how the material will bend the light rays
    #  value must be between <0,1] (total reflection/mirror)
    def SetReflection(self, value):
        """ generated source for method SetReflection """
        self.reflection = value

    def GetRefractionIn(self):
        """ generated source for method GetRefractionIn """
        return self.refractionIn

    def SetRefractionIn(self, value):
        """ generated source for method SetRefractionIn """
        self.refractionIn = value

    def GetRefractionOut(self):
        """ generated source for method GetRefractionOut """
        return self.refractionOut

    def SetRefractionOut(self, value):
        """ generated source for method SetRefractionOut """
        self.refractionOut = value

    #  indicates that the material has a texture and therefore the exact
    #  u,v coordinates are to be calculated by the element
    #  and passed on in the GetColor function
    def HasTexture(self):
        """ generated source for method HasTexture """
        return False

    #  retrieves the actual color of the material
    def GetColor(self):
        """ generated source for method GetColor """
        return self.color

    def SetColor(self, colorp):
        """ generated source for method SetColor """
        self.color = colorp

